hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/DeletedBlockLogImpl.java

```diff
@@ -344,6 +346,9 @@ public DatanodeDeletedBlockTransactions getTransactions(
       try (TableIterator<Long,
           ? extends Table.KeyValue<Long, DeletedBlocksTransaction>> iter =
                deletedBlockLogStateManager.getReadOnlyIterator()) {
+        if (nextProcessTransactionId != -1) {
+          iter.seek(nextProcessTransactionId);
```

Comment:
There is a possibility that the `nextProcessTransactionId` is deleted from the table. In that case comparing `nextProcessTransactionId` with `keyValue.getKey()` will make no sense.